test-util/jvm-test/src/main/java/com/pronoidsoftware/testutil/jvmtest/core/data/networking/MockHttpClientEngineFactory.kt

```diff
@@ -0,0 +1,72 @@
+package com.pronoidsoftware.testutil.jvmtest.core.data.networking
+
+import com.pronoidsoftware.testutil.jvmtest.TestConstants
+import io.ktor.client.engine.HttpClientEngine
+import io.ktor.client.engine.mock.MockEngine
+import io.ktor.client.engine.mock.respond
+import io.ktor.client.engine.mock.toByteArray
+import io.ktor.http.ContentType
+import io.ktor.http.HttpStatusCode
+import io.ktor.http.headersOf
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.jsonObject
+import kotlinx.serialization.json.jsonPrimitive
+
+class MockHttpClientEngineFactory {
+    private val responseHeaders =
+        headersOf("Content-Type" to listOf(ContentType.Application.Json.toString()))
+
+    fun create(): HttpClientEngine {
+        return MockEngine { request ->
+            val body = request.body.toByteArray().decodeToString()
+            val json = Json.parseToJsonElement(body).jsonObject
+            var content = ""
+            var status = HttpStatusCode.OK
```

Comment:
I used to encounter issues with this Ktor mock client once I needed control over the dispatchers, since it doesn't let you pass your own one by default. I ended up creating my own mock client based on the official one:
```
class TestMockEngine(
    override val dispatcher: CoroutineDispatcher,
    private val mockEngineConfig: MockEngineConfig
) : HttpClientEngine {
    val mockEngine: MockEngine = MockEngine(mockEngineConfig)

    override val coroutineContext: CoroutineContext = mockEngine.coroutineContext + dispatcher

    override val config: HttpClientEngineConfig
        get() = mockEngineConfig

    @InternalAPI
    override suspend fun execute(data: HttpRequestData): HttpResponseData {
        return withContext(coroutineContext) {
            mockEngine.execute(data)
        }
    }

    override fun close() {
        coroutineContext.cancel()
    }
}
```
If it works for you as it is, no reason to change this, but just in case you encounter issues in future